### DISCUSSION TITLE
Custom container

### DISCUSSION THREAD

**Discussion by docjyJ** [nextcloud/all-in-one#4822] [Ideas] [answered]
Is it possible to add your own container?

For example, mount a json file my_container.json

**Comment by szaimen** [comment, marked as answer]
Hi, for what do you need this? 

But yes, it should be possible to mount a custom folder say `my_custom` into the container inside  `/var/www/docker-aio/community-containers/` in the container and also provide a `my_custom.json` in the folder. then you can enable the container via `--env AIO_COMMUNITY_CONTAINERS="my_custom"`.

**Reply by docjyJ** [reply]
Yes, I didn't think of that, but yes, it must work. Thanks a lot!

I want to add a caddy version with my custom configuration.

Could the use of local build containers be a problem?

**Reply by szaimen** [reply]
> I want to add a caddy version with my custom configuration.

You could have a look at https://github.com/nextcloud/all-in-one/tree/main/community-containers/caddy. There you can also add custom configs. See the note pretty far down.

> Could the use of local build containers be a problem?

In general yes, for example update checks are of course not going to work...

**Reply by docjyJ** [reply]
> You could have a look at https://github.com/nextcloud/all-in-one/tree/main/community-containers/caddy. There you can also add custom configs. See the note pretty far down.

Oh great! But it requires copying a file, too bad we can't add a volume directly to Caddy.
I prefer to run my own container.


> In general yes, for example update checks are of course not going to work...

😔 I'm thinking... I should make a list of features that I need for my server.

**Reply by szaimen** [reply]
> But it requires copying a file, too bad we can't add a volume directly to Caddy.

The files are stored in a docker volume so not sure what is missing?

**Reply by szaimen** [reply]
> I'm thinking... I should make a list of features that I need for my server.

Yeah, makes sense I guess

**Reply by docjyJ** [reply]
> The files are stored in a docker volume so not sure what is missing?

I will look at this when stalwart is implemented

**Reply by szaimen** [reply]
Feel free to post the list here. I am always interested in what features people need :)

**Reply by docjyJ** [reply]
Yes,

I'm looking for nocodb. I migrated from NC Deck to Table, but it's missing features like Kanban views.

See: https://github.com/nextcloud/tables/issues/103

Currently, I use a docker compose with:
- [Caddy Security](https://github.com/greenpau/caddy-security)
- [OpenLdap](https://github.com/osixia/docker-openldap) & [phpLDAPadmin](https://github.com/osixia/docker-phpLDAPadmin) *LLDAP missing features :(*
- NC + Stalwart

I'm also looking for a Git solution, probably GitLab.
And OAth provider...
And a much too long task list (that’s why NocoDb ^^).

**Reply by docjyJ** [reply]
ps: My goal, to use only open-source, with the same quality as the Google/Microsoft/Apple ecosystem.

**Reply by szaimen** [reply]
Interesting! Thanks for sharing! :)

**Reply by docjyJ** [reply]
```json
{
    "aio_services_v1": [
        {
            "container_name": "nextcloud-aio-nocodb",
            "display_name": "NocoDB",
            "documentation": "https://github.com/nextcloud/all-in-one/tree/main/community-containers/nocodb",
            "image": "docjyj/aio-nocodb",
            "image_tag": "%AIO_CHANNEL%",
            "internal_port": "10028",
            "restart": "unless-stopped",
            "ports": [
                {
                    "ip_binding": "%APACHE_IP_BINDING%",
                    "port_number": "10028",
                    "protocol": "tcp"
                }
            ],
            "environment": [
                "NC_AUTH_JWT_SECRET=%NOCODB_JWT_SECRET%",
                "NC_PUBLIC_URL=https://table.%NC_DOMAIN%/",
                "NC_ADMIN_EMAIL=admin@noco.db",
                "NC_ADMIN_PASS=%NOCODB_USER_PASS%",
                "PORT=10028",
                "NC_DISABLE_TELE=true"
            ],
            "secrets": [
                "NOCODB_JWT_SECRET",
                "NOCODB_USER_PASS"
            ],
            "volumes": [
                {
                    "source": "nextcloud_aio_nocodb",
                    "destination": "/usr/app/data",
                    "writeable": true
                }
            ],
            "backup_volumes": [
                "nextcloud_aio_nocodb"
            ]
        }
    ]
}
```
Volume : `<path/to.json>:/var/www/docker-aio/community-containers/my_nocodb/my_nocodb.json`
Env: `AIO_COMMUNITY_CONTAINERS=<other> my_nocodb`
Repo: https://github.com/docjyJ/aio-nocodb

Publish and functional on my server!

 *Note: When Nextcloud Tables will be at the same level, I will stop keeping up to date because it will be useless*

Do I make PR?

**Reply by szaimen** [reply]
> Publish and functional on my server!

Nice!

> _Note: When Nextcloud Tables will be at the same level, I will stop keeping up to date because it will be useless_

I see...

> Do I make PR?

This is of course up to you but since it is already ready, I would say yes, go for it :)

I guess you could state this disclaimer in the readme though to transparently inform people directly. Then if the day should come, someone from the community can take over if they need it :)

**Reply by szaimen** [reply]
Btw, @docjyJ since you dont seem to change anything insjde the container, you could also use the nocodb/nocodb image directly. See e.g. https://github.com/nextcloud/all-in-one/blob/cbe0a89676001bff86cf2ee9c7aea0767ce21b0d/community-containers/vaultwarden/vaultwarden.json#L7

**Reply by szaimen** [reply]
Or am I missing something?

**Reply by docjyJ** [reply]
> I guess you could state this disclaimer in the readme though to transparently inform people directly. Then if the day should come, someone from the community can take over if they need it :)

Everyone will want to change for an intricate tool directly in NextCloud. :)

> Btw, @docjyJ since you dont seem to change anything insjde the container, you could also use the nocodb/nocodb image directly. See e.g. 

I want to be informed of updates in case of braking change. But, I can change it in my PR.
